### ISSUE TITLE
Add Explanation for why sudo is requested in installer (configure.sh)

### ISSUE THREAD

**Issue by arnuschky** [OpenBazaar/OpenBazaar#1266]
Quite dodgy to ask for the sudo password without at least telling the user what's it for.

**Comment by hoffmabc** [comment]
Dodgy is a debatable term since you can simply look at the script if you're apprehensive about doing it, but I hear your concern. We can add a description of why it's needed in the installer.

**Comment by gubatron** [comment]
@arnuschky how about a patch with the prompt that explains why?

**Comment by arnuschky** [comment]
I think it would be better to show the required commands to the user for execution. That way, a user can also make decisions about required upgrades, conflicts, etc.

If an all-in-one patch is preferred, there should be at least some sort of prompt similar to what you're doing already in `installArch`.

I can provide a patch if need be, sure.

**Comment by Renelvon** [comment]
@arnuschky Many people complain about the `sudo` in the installation but none complain about the hidden bitcoin miner inside OpenBazaar!

Meaning: If you trust that *running* the source code will not abuse your machine in one of the myriad ways possible, how come you don't trust that the installation script does only what it is supposed to do?

You should rethink your threat model; if you believe that installing or running OpenBazaar may compromise your machine's security (remember, we are still in Beta), then you should install and run the program inside a sandboxed VM.

[There is no bitcoin miner, of course.]

**Comment by andre-amorim** [comment]
or perhaps a docker container, anyway at least in my system the 'hidden'
bitcoin miner process will need to compete ( race condition ?! ) to several
others miners process LOL Good Luck ! cycle clock robberies LOL ;)

:.  Andre Amorim      .:

Key-ID:

A70B444E

Fingerprint:

FAAD3B6556A8877B938EAB2FC05E2CB0A70B444E

**Comment by arnuschky** [comment]
@Renelvon That's precisely the point: I don't trust software off the internet, and therefore I run it under a non-privileged user. If that software then wants to have superuser rights, I find that dubious. But yeah, maybe it's just me and nobody else cares...

**Comment by hoffmabc** [comment]
I guess this is more of an argument of whether or not we think it's worth annoying most users and making them do manual steps vs. just looking at the open source code if they are concerned about these "dubious" rights elevations. 

Personally I think it's very easy to just scan through the script and see what's what. I'm not sure about the average user, but I am sure they probably won't like being told to go off script and do some stuff. The point is to balance making it seamless and simple for the user to install and secure and trustworthy. I think this change request tips the scales too far in the security side.

**Comment by SamPatt** [comment]
The issue here isn't about using sudo or not, it's about whether or not to add an explanation. Assuming such an explanation is simple to add, I'm not opposed to it.

**Comment by chrysophylax69** [comment]
hi all ...

i am no coder - and have been interested in the package for quite some time ...

in the event that an installer wants my su password - and as far as a 'standard' user would go - why the hell would i want to even consider looking into the script itself? ... an average user just doesnt do such things ...

a simple 'this is why we need the su password for the installer to continue' would be sufficient - if in fact the target user IS your 'average' user ...

not everyone is that inquisitive that they will scour the scripts and source code to 'feel safe' that the package is not doing anything harmful ... in my experience - most will just say no and walk away from the package altogether ...

#crysx

**Comment by arnuschky** [comment]
It's a decision between security, convention, and convenience. The PR implements only a message, but executes still the package install via sudo. Only tested ubuntu.

**Comment by gubatron** [comment]
>The issue here isn't about using sudo or not, it's about whether or not to add an explanation. Assuming such an explanation is simple to add, I'm not opposed to it.

@SamPatt thinks the same as I, just send a patch with the explanation prompt.

If an user can't trust this script, they're free to install all the dependencies manually, also updating the old wiki page on how to install OpenBazaar manually would be welcome.

TL;DR; I'll repeat, how about a patch with the explanation prompt.

**Comment by hoffmabc** [comment]
yeah we just need to make sure the PR addresses all the OSs necessary @gubatron then we can merge it.

**Comment by gubatron** [comment]
very cool patch btw
